admin/admin.go

```diff
@@ -0,0 +1,508 @@
+type GetPhonesResult struct {
+	duoapi.StatResult
+	Response []Phone
+}
+
+// GetPhones calls GET /admin/v1/phones
+// See https://duo.com/docs/adminapi#phones
+func (c *Client) GetPhones(options ...func(*url.Values)) (*GetPhonesResult, error) {
+	params := url.Values{}
+	for _, o := range options {
+		o(&params)
+	}
+
+	_, body, err := c.SignedCall(http.MethodGet, "/admin/v1/phones", params, duoapi.UseTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &GetPhonesResult{}
+	err = json.Unmarshal(body, result)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// GetPhoneResult models responses containing a single phone.
+type GetPhoneResult struct {
+	duoapi.StatResult
+	Response Phone
+}
+
+// GetPhone calls GET /admin/v1/phones/:phone_id
+// See https://duo.com/docs/adminapi#retrieve-phone-by-id
+func (c *Client) GetPhone(phoneID string) (*GetPhoneResult, error) {
+	path := fmt.Sprintf("/admin/v1/phones/%s", phoneID)
+
+	_, body, err := c.SignedCall(http.MethodGet, path, nil, duoapi.UseTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &GetPhoneResult{}
+	err = json.Unmarshal(body, result)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// Token methods
+
+// GetTokensTypeAndSerial sets the optional type and serial parameters for a GetTokens request.
+func GetTokensTypeAndSerial(typ, serial string) func(*url.Values) {
+	return func(opts *url.Values) {
+		opts.Set("type", typ)
+		opts.Set("serial", serial)
+	}
+}
+
+// GetTokensResult models responses containing a list of tokens.
+type GetTokensResult struct {
+	duoapi.StatResult
+	Response []Token
+}
+
+// GetTokens calls GET /admin/v1/tokens
+// See https://duo.com/docs/adminapi#retrieve-hardware-tokens
+func (c *Client) GetTokens(options ...func(*url.Values)) (*GetTokensResult, error) {
+	params := url.Values{}
+	for _, o := range options {
+		o(&params)
+	}
+
+	_, body, err := c.SignedCall(http.MethodGet, "/admin/v1/tokens", params, duoapi.UseTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &GetTokensResult{}
+	err = json.Unmarshal(body, result)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// GetTokenResult models responses containing a single token.
+type GetTokenResult struct {
+	duoapi.StatResult
+	Response Token
+}
+
+// GetToken calls GET /admin/v1/tokens/:token_id
+// See https://duo.com/docs/adminapi#retrieve-hardware-tokens
+func (c *Client) GetToken(tokenID string) (*GetTokenResult, error) {
+	path := fmt.Sprintf("/admin/v1/tokens/%s", tokenID)
+
+	_, body, err := c.SignedCall(http.MethodGet, path, nil, duoapi.UseTimeout)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &GetTokenResult{}
+	err = json.Unmarshal(body, result)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// U2F token methods
+
+// GetU2FTokensLimit sets the optional limit parameter for a GetU2FTokens request.
+func GetU2FTokensLimit(limit uint64) func(*url.Values) {
```

Comment:
minor: It would be nice to find a way to implement this just once, and share among all the API call wrappers. It should be generic enough to do so, since all the optional functions have the same signature.

Reply:
(and same for offset)

Reply:
Good call! Done.